File: 백준/Silver/9291. 스도쿠 채점/스도쿠 채점.java
import java.io.*;
import java.util.StringTokenizer;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        StringTokenizer st;
        for (int tc = 1; tc <= T; tc++) {
            int[][] board = new int[9][9];
            for (int i = 0; i < 9; i++) {
                st = new StringTokenizer(br.readLine());
                for (int j = 0; j < 9; j++) {
                    board[i][j] = Integer.parseInt(st.nextToken());
                }
            }
            if (tc < T) {
                String tmp = br.readLine();
            }

            boolean result = check(board);
            if (result) {
                System.out.printf("Case " + tc + ": CORRECT\n");
            } else {
                System.out.printf("Case " + tc + ": INCORRECT\n");
            }
        }
    }

    static boolean check(int[][] arr) {
        for (int i = 0; i < 9; i++) {
            boolean[] visitedSero = new boolean[10];
            boolean[] visitedGaro = new boolean[10];
            for (int j = 0; j < 9; j++) {
                if (visitedSero[arr[i][j]]) {
                    return false;
                } else {
                    visitedSero[arr[i][j]] = true;
                }

                if (visitedGaro[arr[j][i]]) {
                    return false;
                } else {
                    visitedGaro[arr[j][i]] = true;
                }

                boolean[] visitedThree = new boolean[10];
                if (i % 3 == 0 && j % 3 == 0) {
                    for (int k = 0; k < 3; k++) {
                        for (int p = 0; p < 3; p++) {
                            if (visitedThree[arr[i+k][j+p]]) {
                                return false;
                            } else {
                                visitedThree[arr[i+k][j+p]] = true;
                            }
                        }
                    }
                }
            }
        }
        return true;
    }
}

File: 백준/Silver/9291. 스도쿠 채점/스도쿠 채점.py
def check(arr):
    for i in range(9):
        visited_garo = [False] * 10
        visited_sero = [False] * 10

        for j in range(9):
            # 가로 확인
            if visited_garo[board[i][j]]:
                return False
            else:
                visited_garo[board[i][j]] = True

            # 세로 확인
            if visited_sero[board[j][i]]:
                return False
            else:
                visited_sero[board[j][i]] = True

            # 3 x 3 확인
            if i % 3 == 0 and j % 3 == 0:
                visited_three = [False] * 10
                for k in range(3):
                    for p in range(3):
                        if visited_three[board[i+k][j+p]]:
                            return False
                        else:
                            visited_three[board[i+k][j+p]] = True
    return True

T = int(input())
for tc in range(1, T+1):
    board = [list(map(int, input().split())) for _ in range(9)]
    if tc != T:
        tmp = list(input())

    result = check(board)
    if result:
        print(f'Case {tc}: CORRECT')
    else:
        print(f'Case {tc}: INCORRECT')